vortex-array/src/array/varbin/builder.rs

```diff
@@ -11,7 +11,7 @@ use crate::IntoArrayData;
 
 pub struct VarBinBuilder<O: NativePType> {
     offsets: Vec<O>,
-    data: BytesMut,
+    data: VortexBytesMut,
```

Comment:
We still have all of our Vec-based allocations not using aligned buffers... I'll have a think about how to pull this out, but we likely need something akin to Arrow's ScalarBuffer<T>.